cpp/src/arrow/engine/substrait/ext_test.cc

```diff
@@ -285,5 +340,55 @@ TEST(ExtensionIdRegistryTest, RegisterNestedFunctions) {
   }
 }
 
+TEST(ExtensionIdRegistryTest, RegisterNestedAggregates) {
```

Comment:
What is this testing?

Reply:
All but the last test I added (this one included) are for aggregate registration that were just missing. They are not specifically related to the main-code change here, but IMHO they are needed for basic coverage. They follow the existing tests for function registration.

The last test I added is the one that specifically covers the main-code change here.

Reply:
Right. But I wonder what specially are you trying to test in this one? It is not clear what "NestedAggregates" means here

Reply:
"Nested" refers to a nested registry, as opposed to the default one, in which the items are registered. This is also the case in the existing "NestedFunctions" and "NestedTypes" tests.

Reply:
What do you mean "nested registry"?  looks like registry1 and registry2 are orthgonal (we don't see them being nested together maybe I missed sth)?

Reply:
The invocation `MakeExtensionIdRegistry()`, which is used in `Nested` test-cases, [makes a nested registry on top of the default one](https://github.com/apache/arrow/blob/99ef3aada1d1589604da8905f3c9100ff1c997e9/cpp/src/arrow/engine/substrait/util.cc#L64).

Reply:
I see.